eppo/src/main/java/cloud/eppo/android/dto/deserializers/RandomizationConfigResponseDeserializer.java

```diff
@@ -0,0 +1,194 @@
+    private static final String TAG = logTag(EppoClient.class);
+
+    private final EppoValueAdapter eppoValueAdapter = new EppoValueAdapter();
+
+    @Override
+    public RandomizationConfigResponse deserialize(JsonElement rootElement, Type type, JsonDeserializationContext context) throws JsonParseException {
+        ConcurrentHashMap<String, FlagConfig> flags = new ConcurrentHashMap<>();
+        RandomizationConfigResponse configResponse = new RandomizationConfigResponse();
+        configResponse.setFlags(flags); // Default to a response with an empty map of flags
+
+        if (rootElement == null || !rootElement.isJsonObject()) {
+            Log.w(TAG, "no top-level JSON object");
+            return configResponse;
+        }
+
+        JsonObject rootObject = rootElement.getAsJsonObject();
+        JsonElement flagsElement = rootObject.get("flags");
+        if (flagsElement == null) {
+            Log.w(TAG, "no root-level flags property");
+            return configResponse;
+        }
+
+        JsonObject flagsObject = flagsElement.getAsJsonObject();
+        for (Map.Entry<String, JsonElement> flagEntry : flagsObject.entrySet()) {
+            String flagKey = flagEntry.getKey();
+            FlagConfig flagConfig = deserializeFlag(flagEntry.getValue(), type, context);
+            flags.put(flagKey, flagConfig); // Note this is adding to the map already plugged into configResponse
+        }
+
+        return configResponse;
+    }
+
+    private FlagConfig deserializeFlag(JsonElement jsonElement, Type type, JsonDeserializationContext context) {
+        JsonObject flagObject = jsonElement.getAsJsonObject();
+        int subjectShards = flagObject.get("subjectShards").getAsInt();
+        boolean enabled = flagObject.get("enabled").getAsBoolean();
+        Map<String, String> typedOverrides = deserializeTypedOverrides(flagObject.get("typedOverrides"));
+        List<TargetingRule> rules = deserializeTargetingRules(flagObject.get("rules"), type, context);
+        Map<String, Allocation> allocations = deserializeAllocations(flagObject.get("allocations"), type, context);
+
+        FlagConfig flagConfig = new FlagConfig();
+        flagConfig.setEnabled(enabled);
+        flagConfig.setSubjectShards(subjectShards);
+        flagConfig.setTypedOverrides(typedOverrides);
+        flagConfig.setRules(rules);
+        flagConfig.setAllocations(allocations);
+        return flagConfig;
+    }
+
+    private Map<String, String> deserializeTypedOverrides(JsonElement jsonElement) {
+        Map<String, String> typedOverrides = new HashMap<>();
+        if (jsonElement == null) {
+            return typedOverrides;
+        }
+
+        for (Map.Entry<String, JsonElement> typedOverridesEntry : jsonElement.getAsJsonObject().entrySet()) {
+            typedOverrides.put(typedOverridesEntry.getKey(), typedOverridesEntry.getValue().getAsString());
```

Comment:
`getValue()` shouldn't ever be null here right?

Reply:
It _could_ be, as this is JSON, so it's certainly possible. However, if we have a `typedOverrides,` we're going to count on it being properly formatted. If not, this will throw a `JsonParseException,` which is what we want in this situation. Both places we parse are surrounded by a higher-level `try`-`catch` to handle that exception appropriately.